src/main/java/edu/hm/hafner/grading/ToolConfiguration.java

```diff
@@ -85,6 +93,14 @@ public String getIcon() {
         return StringUtils.defaultString(icon);
     }
 
+    public String getSourcePath() {
+        return StringUtils.defaultString(sourcePath);
+    }
+
+    public Scope getScope() {
+        return Scope.fromString(scope);
```

Comment:
Without the check I get a `SOE`
```suggestion
        if (StringUtils.isBlank(scope)) {
            return Scope.PROJECT;
        }
        return Scope.fromString(scope);
```